File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(std::vector<int>& arr) {
        std::unordered_map<int, int> count;
        std::unordered_map<int, int> unique;

        for (int v : arr) {
            count[v] = count[v] + 1;
        }

        for (const auto& entry : count) {
            int v = entry.second;
            if (unique.count(v) > 0) {
                return false;
            }
            unique[v] = 1;
        }

        return true;
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        a = dict()
        l=[]
        for i in arr:
            if i not in a:
                a[i]=1
            else:
                a[i]+=1
        for i in a:
            l.append(a[i])
        n=len(l)
        s=set(l)
        m=len(s)
        if n==m:
            return True
        else:
            return False